### ISSUE TITLE
Operation process

### ISSUE THREAD

**Issue by LJJcodeing** [heleidsn/px4_avoidance_airsim#2]
Ok, it's me again, nice to have a cordial conversation with you again >^<.
Now the question is, so far I've finished building the airsim environment, ros, and testing the airsim ros package, and have likewise finished building the px4 environment, so how do I use this fusion algorithm next? Or is there any need for caktkin_make?
And can you share the configuration information about the setting.json file of this airsim project?
Looking forward to your guidance, appreciate it b(￣▽￣)d.
![Uploading 屏幕截图 2024-04-08 161637.png…]()

**Comment by heleidsn** [comment]
Hi, maybe you can follow the Running in simulation part in the readme file. In short, you have to
1) launch the simulation 
2) launch the planning algorithms
3) enjoy

Btw, your screenshots are not showing properly. Please upload a new one.

**Comment by LJJcodeing** [comment]
Okay, now I have figured out how to start the execution, but the first error came. 
When I was in roslaunch, which was executing the first start_simulator.launch file, it couldn't find my airsim'ros_pkgs. 
The error message showed that the search path was incorrect, including the one in px4 that followed.
 So where should this path problem be solved?

**Comment by LJJcodeing** [comment]
the error such as this:

Resource not found: airsim_ros_pkgs
ROS path [0]=/opt/ros/melodic/share/ros
ROS path [1]=/opt/ros/melodic/share
The traceback for the exception was written to the log file


btw,your readme file has something wrong ,the " roslaunch px4_avoidance_airsim start_simulation.launch"should be"roslaunch px4_avoidance_airsim start_**simulator**.launch

**Comment by LJJcodeing** [comment]
![error](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/69051a93-4899-4c67-b0b0-c9b0936e42b0)

**Comment by LJJcodeing** [comment]
![error2](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/17f24bd1-6a0b-4eca-83f5-b4fe0df7840d)

**Comment by LJJcodeing** [comment]
emmm，There seems to be some issues with this workspace as well    QAQ
![error2](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/2f90239b-b2ae-40f6-8fbe-a17183c12d8e)
appreciate so much

**Comment by heleidsn** [comment]
Maybe you have to `source devel/setup.bash` before launch in terminal

**Comment by LJJcodeing** [comment]
Yes, I've tried this before, but it doesn't seem to be very useful
![error3](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/9fb42df5-241a-42e3-bc88-c8066c9540bb)

**Comment by LJJcodeing** [comment]
Alight，finally i solve this problem，i create  a new "catikin_ws"  file to add the ros path.
However,a new issue  came .
Did i need the default px4.board?
I can't pip the  kconfiglib,
if the repo don't need ,i think i can delete this dependences to solve this problem.
(Actually, I have just started working in this field and often ask silly questions. I hope you don't mind)
Iooking forward your relpy.thanks
![error4](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/56b83fde-a07f-44e2-8f31-d287909e23f2)

**Comment by heleidsn** [comment]
Hi, just saw your issues. 

As I didn't work on this repo for a period of time, I am not very sure if kconfiglib is necessary. However, the problem related to Python package is mainly caused by your Python path and configuration. Maybe you should check the Python version used for catkin build. 

Anyway, can you run PX4-avoidance individually?

**Comment by LJJcodeing** [comment]
Thank you for your reply

As you expected, I found that I was using Python 2.7 during compilation and downloaded kconfiglib to Python 3.8, so the compilation failed. After I solved this problem according to your prompts, other problems occurred. So, I took out catkin_ws separately for px4 and exported the ROS path, and **I was able to successfully start simulator. launch** (Absolutely good news!)

But when starting "fast planner. launch", some nodes failed to start, and so far I don't have any clue. Do you have any good suggestions?
Also, I haven't tried launching px4 avoidance separately yet

Thanks a lot!

**Comment by LJJcodeing** [comment]
this is the new error and my src file
![error6](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/779ec748-d8b8-40b0-9da5-c0369d960b6f)
![src](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/f296a2d7-ebbd-43c4-a5fb-091f47ef3335)

**Comment by LJJcodeing** [comment]
such as this
![error7](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/372b96ad-9bd5-4c99-9f7b-ef1b7751e7fc)

**Comment by heleidsn** [comment]
From my experience, it's better to compile all the ros packages using python3. It's hard to say what's wrong from your screenshot now. However, it's better to run px4_avoidance and fast planner individually before using this repo.  Good luck.

**Comment by LJJcodeing** [comment]
I used python3.6,but tf2 has some error.
Actually ,I can run airsim+px4+mavros smoothly,but only the fast planner could't recive image and the map
have you have any suggestions?

**Comment by LJJcodeing** [comment]
![2024-04-29 09-10-44 的屏幕截图](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/46ae5ba0-d651-4bff-ae57-e6669309478b)

**Comment by LJJcodeing** [comment]
![2024-04-29 09-11-27 的屏幕截图](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/1b5602eb-be99-494a-bc39-96c27fee07fd)

**Comment by LJJcodeing** [comment]
as you can see the drone can't be contrl by fast planner

**Comment by LJJcodeing** [comment]
could you please provide your "airsim setting json",maybe i got sone wrong from there

**Comment by LJJcodeing** [comment]
![2024-04-30 09-42-28 的屏幕截图](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/898c6fc2-5ef2-441d-8a9b-6d33bc27b82e)
Anyway,it recived .
But,why can't i control it ,what should i do?
is there any commander should i konw,or just ues  **2Dnav goal**?

**Comment by LJJcodeing** [comment]
![2024-04-30 09-43-14 的屏幕截图](https://github.com/heleidsn/px4_avoidance_airsim/assets/132959941/b39d7b22-bfcb-4e7b-a9cc-2b23bcf87c50)
sorry to bother you
thanks in advance

**Comment by heleidsn** [comment]
From your screenshot, I find there is a tf2 related issue. Maybe you should check with it first. This is mainly caused by the python version and tf2 ros package. Maybe you have to build tf2 with python3 in your workspace.
